Route/CmsPageRouter.php

```diff
@@ -207,7 +210,25 @@ protected function generateFromPage(PageInterface $page, array $parameters = arr
             throw new \RuntimeException(sprintf('Page "%d" has no url or customUrl.', $page->getId()));
         }
 
-        return $this->decorateUrl($url, $parameters, $referenceType);
+        if (!$this->context instanceof SiteRequestContext) {
+            @trigger_error(
+                'The context should implement SiteRequestContextInterface on the next release',
+                E_USER_DEPRECATED
+            );
+
+            return $this->decorateUrl($url, $parameters, $referenceType);
+        }
```

Comment:
How would you suggest we fix this? Maybe add another IF inside to check if the SiteRequestContext implements the interface and if not throw the exception?